src/bank.rs

```diff
@@ -212,6 +214,15 @@ impl Bank {
         }
     }
 
+    fn check_last_id_age(last_id_queue: &VecDeque<Hash>, last_id: Hash) -> usize {
+        for (i, id) in last_id_queue.iter().enumerate() {
```

Comment:
How about?

```rust
fn get_last_id_age(last_ids: &VecDeque<Hash>, last_id: &Hash) -> Option<usize> {
    last_ids.iter().rposition(|x| x == last_id).map(|i| last_ids.len() - i)
}
```

Reply:
this seems even harder to grok.  its obfuscating the `last_ids.len() - i`